services/src/main/java/com/linkedin/kmf/common/TopicFactory.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.common;
+
+import java.util.Map;
+
+
+/**
+ * Constructs the monitored topic if it does not exist.
+ *
+ * Implementations of this class should have a public, no-arg constructor.
+ *
+ */
+public interface TopicFactory {
+
+  /**
+   * Creates the specified topic if it does not exist.
+   * @param zkUrl zookeeper connection url
+   * @param topic topic name
+   * @param replicationFactor the replication factor for the topic
+   * @param partitionToBrokerRatio This is multiplied by the number brokers to compute the number of partitions in the topic.
+   * @param config additional configuration parameters passed in from the Kafka Monitor configuration.
+   * @return The number of partitions for the specified topic.
+   */
+  public int createTopic(String zkUrl, String topic, int replicationFactor,
```

Comment:
The produce.topic.topicCreationEnabled is to let Kafka monitor create the topic if it does not exist.  Presumably the user can just leave this set to true and should not get an exception the second time Kafka monitor is run.  I would rather just rename this method to createTopicIfNotExist().  Am I misunderstanding something?

The user can still create the monitored topic outside of Kafka monitor and not having the expected replication factor match the existing replication factor seems to be something that can happen.